install/cupy_builder/_command.py

```diff
@@ -28,40 +29,54 @@ def filter_files_by_extension(
 def compile_device_code(
         ctx: Context,
         ext: setuptools.Extension
-) -> Tuple[List[str], List[str]]:
+) -> Tuple[List[str], List[str], List[concurrent.futures.Future]]:
     """Compiles device code ("*.cu").
 
     This method invokes the device compiler (nvcc/hipcc) to build object
     files from device code, then returns the tuple of:
     - list of remaining (non-device) source files ("*.cpp")
     - list of compiled object files for device code ("*.o")
+    - list of future objects representing nvcc compilation tasks
     """
     sources_cu, sources_cpp = filter_files_by_extension(
         ext.sources, '.cu')
     if len(sources_cu) == 0:
         # No device code used in this extension.
-        return ext.sources, []
+        return ext.sources, [], []
 
     if sys.platform == 'win32':
         compiler = DeviceCompilerWin32(ctx)
     else:
         compiler = DeviceCompilerUnix(ctx)
 
     objects = []
+    futures = []
     for src in sources_cu:
         print(f'{ext.name}: Device code: {src}')
         obj_ext = 'obj' if sys.platform == 'win32' else 'o'
         # TODO(kmaehashi): embed CUDA version in path
```

Comment:
Future work to myself: we should embed (maybe a hash of) CUPY_NVCC_GENERATE_CODE to path as well.

Reply:
Yes I think this is a better solution that checking `settings['depends']`.